src/rgw/s3select.h

```diff
@@ -0,0 +1,852 @@
+#ifndef __S3SELECT__
+#define __S3SELECT__
+
+#include <boost/spirit/include/classic_core.hpp>
+#include <iostream>
+#include <string>
+#include <list>
+#include "s3select_oper.h"
+#include <boost/function.hpp>
+#include <boost/bind.hpp>
+#include <functional>
+
+using namespace std;
+using namespace BOOST_SPIRIT_CLASSIC_NS;
+#define _DEBUG_TERM {string  token(a,b);std::cout << __FUNCTION__ << token << std::endl;}
+
+/// AST builder
+
+class s3select_projections {
+
+    private:
+        list<base_statement *> m_projections;
+    
+    public:
+        bool is_aggregate()
+        {
+            //TODO iterate on projections , and search for aggregate
+            //for(auto p : m_projections){}
+
+	    return false;
+        }
+
+        bool semantic()
+        {
+            //TODO check aggragtion function are not nested 
+            return false;
+        }
+
+        list<base_statement *> * get()
+        {
+            return &m_projections;
+        }
+
+};
+
+struct actionQ
+{
+
+    list<mulldiv_operation::muldiv_t> muldivQ;
```

Comment:
again, would love to avoid list if we can

Reply:
same as above.